interfaces/IBF-dashboard/src/app/services/map.service.ts

```diff
@@ -293,21 +293,46 @@ export class MapService {
     waterPoints: GeoJSON.FeatureCollection,
   ) {
     const isLoading = waterPoints ? false : true;
-    this.addLayer({
-      name: IbfLayerName.waterpoints,
-      label: IbfLayerLabel.waterpoints,
-      type: IbfLayerType.point,
-      group: IbfLayerGroup.point,
-      description: this.getPopoverText(layer),
-      active: this.adminLevelService.activeLayerNames.includes(
-        IbfLayerName.waterpoints,
-      ),
-      show: true,
-      data: waterPoints,
-      viewCenter: false,
-      order: 2,
-      isLoading,
-    });
+
+    // HACK: conditionally add waterpoints layer because it has a loading logic (see loadWaterPointsLayer)
+    // which adds the layer to the map asynchronously in portals where it is not configured
+    // multiple factors are at play here:
+    // 1. waterpoints and waterpoints_internal layers are used to identify the same layer
+    //    which breaks the technical constraint of using unique layer names
```

Comment:
I would still suggest to remove this comment. You are right in saying that IF it was also called 'waterpoints', this particular problem would not have appeared. But that is just 'luck' in the end. Say that 'waterpoints_internal' would not have been present in flash-floods, then still the same bug would happen, in the sense that 'waterpoints' would show up in flash-floods when moving quickly from floods to flash-floods. Likewise, I imagine that previously this bug would occur when switching from floods/drought to malaria.

Don't get me wrong. The '2 layers with same label' situation should also be cleared up (new PBI?), but I would suggest to leave it out here for conciseness/clearness of the comment.